### DISCUSSION TITLE
Can BenchmarkDotNet measure CPU consumption?

### DISCUSSION THREAD

**Discussion by yll690** [dotnet/BenchmarkDotNet#2559] [Q&A] [answered]
Hi, I use BenchmarkDotNet for several times, and wonder if the mean times consider CPU consumption. There may be some algorithms run faster but consume much more CPU resources, resulting in poorer performance when running in parallel.  So does BenchmarkDotNet support measuring CPU resource consumption like cycles and processor time? This also helps comparing results across machines.

**Comment by adamsitnik** [comment, marked as answer]
Hi @yll690 

You need to use a profiler for that. We provide few plugins that make it very easy: https://benchmarkdotnet.org/articles/configs/diagnosers.html

**Reply by yll690** [reply]
Thank you, it's very useful
